unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -209,6 +308,10 @@ public boolean equals(final Object other) {
 
   @Override
   public int hashCode() {
-    return Arrays.hashCode(bytes);
```

Comment:
Once we change the hashCode of String, some PySpark doc tests will fail, so I revert the change, let's do it in separate PR.